core/lib/multivm/src/versions/vm_latest/types/internals/transaction_data.rs

```diff
@@ -85,7 +91,7 @@ impl From<Transaction> for TransactionData {
                     value: execute_tx.execute.value,
                     reserved: [
                         should_check_chain_id,
-                        U256::zero(),
+                        should_deploy_contract,
                         U256::zero(),
                         U256::zero(),
                     ],
```

Comment:
@0xVolosnikov Here is the change on the reserved bytes of the transaction where we set the 0 or 1 depending if we want to deploy an EVM contract or not.